### ISSUE TITLE
parkrun HQ have changed the datasource for the event map

### ISSUE THREAD

**Issue by fraz3alpha** [fraz3alpha/running-challenges#174]
Parkrun have recently changed their map, and with it they have changed their source of data for the markers. This data is used in the extension for the country flags as well as the regionnaire challenge.

- The old file was called [geo.xml](https://www.parkrun.org.uk/wp-content/themes/parkrun/xml/geo.xml) and there was a separate one for the junior parkruns.
- The new file is called [events.json](https://images.parkrun.com/events.json)

The new format looks much better for parsing in general, but removes the notion of regions.

### Events

Here is the entry for Winchester:
```
{
"id": 280,
"type": "Feature",
"geometry": {
"type": "Point",
"coordinates": [
-1.310849,
51.069286
]
},
"properties": {
"eventname": "winchester",
"EventLongName": "Winchester parkrun",
"EventShortName": "Winchester",
"countrycode": 97,
"seriesid": 1,
"EventLocation": "North Walls Recreation Ground"
}
}
```
and here is Winchester juniors:
```
{
"id": 1493,
"type": "Feature",
"geometry": {
"type": "Point",
"coordinates": [
-1.304893,
51.054733
]
},
"properties": {
"eventname": "winchester-juniors",
"EventLongName": "Winchester junior parkrun",
"EventShortName": "Winchester juniors",
"countrycode": 97,
"seriesid": 2,
"EventLocation": "Winchester Sports Stadium"
}
}
```

All the events are listed in event number order in an array under:
```
events": {
"type": "FeatureCollection",
"features": [
...
```

The `seriesid` allows the code to filter based on parkrun/junior parkrun, which is useful for us as we can support junior parkrun without having to fetch a second file.

For the countries, we now get:
```
{
"countries": {
"3": {
"url": "www.parkrun.com.au",
"bounds": [
115.074331,
-43.163256,
153.576356,
-12.376205
]
},
...
```

We used to get a single lat/lon coordinate, which is where we placed the flag, but now we get bounds. This no doubt makes scaling the map easier for parkrun HQ, and I'm sure we can find the mid-point to place the flag.

The JSON file is also much easier to parse.

### Summary

For now, the extension is still working because they haven't removed the old file yet, but I suspect that is only a matter of time. 

*Good*
- The new file is easier to parse.
- We get the junior event data for free.

*Neutral*
- Country flags are still possible to calculate from the data as-is.

*Bad*
- We lose the regions, and the ability to calculate them from the data supplied. Someone is going to end up being the arbiter of which region a parkrun is in if we continue to support the regionnaire challenge.

### Update: 20191114

parkrun have finally removed the geo.xml file, which means that the extension is actually broken now. It can't load any events, which means it throws an error in the options page and you can't save your profile anymore.

I guess that has forced our hand, we'll be off to the events.json file, and regionnaire info will have to go away, for a bit at least.

**Comment by MikeDimmick** [comment]
I note that the event pages on wiki.parkrun.com are auto-generated. The Wiki markup comes from the results system. Right at the end, the UK events' pages have a [[Category]] tag for the event's region. For example, https://wiki.parkrun.com/index.php/Category:South_East_England_Events lists all the events in SE England.

Obviously this isn't going to be too easy to parse, but it's an option if all else fails!

**Comment by fraz3alpha** [comment]
We received a copy of an email enquiry that was made to parkrun HQ regarding the future of regions, and the reply said:

> Regions have been removed from our events map.  Regions have no real relevance to the parkrun organisation and the boundaries were often arbitrary, so maintaining regions makes no sense.

I think we have to assume that although the current events have regions, new ones will not. If they don't mean anything to parkrun HQ, I can't see why they would bother. Maybe the proof will be finding the newest one, maybe in a couple of weeks, and seeing what information it has available. Existing planned ones might already have a region assigned, maybe we won't really know for a month or two yet.

**Comment by rboyatt** [comment]
It's a shame about the loss of regions.  Curiously there's currently at least one event in geo.xml that doesn't feature in events.json (a new event that's not started it seems - it does have a region in the xml).

**Comment by marcburrows** [comment]
Cheers for the explanation. The data looks so much nicer, which is great! I agree with parkrun HQ that the regions were so arbitrary, especially those on the outskirts of greater London being included in various compass named regions.

**Comment by fraz3alpha** [comment]
The old geo.xml file has finally disappeared, https://www.parkrun.org.uk/wp-content/themes/parkrun/xml/geo.xml gives a 404 now :(

<img width="1239" alt="Screenshot 2019-11-14 at 21 19 14" src="https://user-images.githubusercontent.com/2786011/68896962-6ef52c80-0724-11ea-8a65-576c93052363.png">

I've got a branch somewhere where I've started moving across, I'll have to finish that in the next week or two if I can

**Comment by StephenRD** [comment]
Thanks for your efforts in migrating to the JSON file.

I don't agree with the description of the regions as arbitrary. Arbitrary: "based on random choice or personal whim, rather than any reason or system". The regions, for the UK at least, are the standard NUTS statistical regions, which in turn are based on counties and the "nations" of the UK, and are entirely systematic, not random or based on personal whim.

They may produce some non-instinctive outcomes (Rickmansworth doesn't feel like East of England, for example), but that doesn't make them arbitrary, nor mean that they didn't have value in practice for some people, even if not for the parkrun organisation or particular (many) individuals. Some reguibs, such as Scotland, Wales, Northern Ireland and (Greater) London have greater emotional affinity than others.

**Comment by fraz3alpha** [comment]
@StephenRD , I am aware of the standard regions which parkrun have based theirs on, but it's not really for us to agree or disagree whether parkrun have used them to arbitrarily split up the parkruns in the UK - the fact is they did, and not always consistently, as it suited them.

If we decided to bring them back, the fact is some parkruns would move region in the UK, compared to what they were before.

Note, these region definitions aren't in use in any other country, and this extension is not exclusive to the UK.

**Comment by lauracowen** [comment]
Fixed in release v0.7.7.
